### PR TITLE
[logging] Guard verbose logs on glog.Verbose

### PR DESCRIPTION
**Pull request by pgavlin**
`glog.Verbose(int)` returns a bool that indicates whether or not logging is enabled at the given level. Check that result prior to actually calling the logger in order to avoid extra work when verbose logging is disabled.

### CODE REVIEW

File: sdk/go/common/util/logging/log.go [hunk 1]
```diff
@@ -60,19 +60,25 @@ type VerboseLogger glog.Verbose
 // Info is equivalent to the global Info function, guarded by the value of v.
 // See the documentation of V for usage.
 func (v VerboseLogger) Info(args ...interface{}) {
-	glog.Verbose(v).Info(FilterString(fmt.Sprint(args...)))
+	if log := glog.Verbose(v); log {
+		log.Info(FilterString(fmt.Sprint(args...)))
+	}
```
Comment (abhinav) [inline sdk/go/common/util/logging/log.go:65]: So `glog.Verbose` doesn't return a boolean. It is a boolean.
And VerboseLogger is a wrapper around it, so VerboseLogger is also a boolean.

I think this is good change in that we can avoid paying for FilterString and Sprintf, but we can shorten this to:

```go
if v {
  glog.Verbose(v).Info(...)
}
```

**Review by abhinav — APPROVED** [review]
Approving either way. Change is a net positive.
Optional shortening is optional.

File: sdk/go/common/util/logging/log.go (hunk 1 above)
Comment (pgavlin) [reply sdk/go/common/util/logging/log.go:65]: > So `glog.Verbose` doesn't return a boolean. It is a boolean. 

Yes, I know--hence the `log := ...; log { log\.Infof }`. "return a boolean", while imprecise, felt a little easier on the reader.

> And VerboseLogger is a wrapper around it, so VerboseLogger is also a boolean.

This is an excellent point, though. I went for the simplest refactoring--lifting the calls to `glog.Verbose` that these methods were making--but what you suggest is even better.

**Comment by pgavlin** [comment]
bors merge
